File: leetcode/minimum-size-subarray-sum.py
from typing import List


class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        min_length = 100001  # see constraints
        length = len(nums)

        l = s = 0

        for i in range(length):
            s += nums[i]

            if s >= target:
                while s >= target:
                    s -= nums[l]
                    l += 1

                min_length = min(min_length, i - l + 2)

        if min_length > 100000:
            min_length = 0

        return min_length

File: leetcode/minimum-size-subarray-sum.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
 public:
    int minSubArrayLen(int target, vector<int>& nums)
    {
        auto size = nums.size();
        int len = INT_MAX;

        int l = 0, sum = 0;

        for (auto r = 0; r < size; ++r) {
            sum += nums[r];

            while (sum >= target) {
                len = min(len, r - l + 1);
                sum -= nums[l];
                ++l;
            }
        }

        return len == INT_MAX ? 0 : len;
    }
};
